tests/foreman/cli/test_host.py

```diff
@@ -1,67 +1,82 @@
 """CLI tests for ``hammer host``."""
-from fauxfactory import gen_string
+from fauxfactory import gen_mac, gen_string
 from nailgun import entities
 from robottelo.cli.base import CLIReturnCodeError
+from robottelo.cli.factory import (
+    make_architecture,
+    make_domain,
+    make_environment,
+    make_medium,
+    make_os,
+)
 from robottelo.cli.host import Host
+from robottelo.cli.medium import Medium
+from robottelo.cli.operatingsys import OperatingSys
 from robottelo.cli.proxy import Proxy
 from robottelo.config import settings
-from robottelo.datafactory import invalid_values_list, valid_data_list
+from robottelo.datafactory import (
+    invalid_values_list,
+    valid_data_list,
+    valid_hosts_list,
+)
 from robottelo.decorators import tier1, tier2
 from robottelo.test import CLITestCase
 
 
-class HostTestCase(CLITestCase):
-    """Host CLI tests."""
+class HostCreateTestCase(CLITestCase):
+    """Tests for creating the hosts via CLI."""
 
     def setUp(self):
         """Find an existing puppet proxy.
 
         Record information about this puppet proxy as ``self.puppet_proxy``.
-
         """
+        super(HostCreateTestCase, self).setUp()
         # Use the default installation smart proxy
         result = Proxy.list()
         self.assertGreater(len(result), 0)
         self.puppet_proxy = result[0]
 
     @tier1
-    def test_positive_create_1(self):
+    def test_positive_create_with_name(self):
         """@test: A host can be created with a random name
 
         @feature: Hosts
 
         @assert: A host is created and the name matches
-
         """
-        host = entities.Host()
-        host.create_missing()
-        result = Host.create({
-            u'architecture-id': host.architecture.id,
-            u'domain-id': host.domain.id,
-            u'environment-id': host.environment.id,
-            u'location-id': host.location.id,  # pylint:disable=no-member
-            u'mac': host.mac,
-            u'medium-id': host.medium.id,
-            u'name': host.name,
-            u'operatingsystem-id': host.operatingsystem.id,
-            u'organization-id': host.organization.id,  # pylint:disable=E1101
-            u'partition-table-id': host.ptable.id,
-            u'puppet-proxy-id': self.puppet_proxy['id'],
-            u'root-pass': host.root_pass,
-        })
-        self.assertEqual(
-            '{0}.{1}'.format(host.name, host.domain.read().name).lower(),
-            result['name'],
-        )
+        for name in valid_hosts_list():
+            with self.subTest(name):
+                host = entities.Host()
+                host.create_missing()
+                result = Host.create({
```

Comment:
Does this create real host? I ask because we dont have rhel content synced here.